internal/history/service.go

```diff
@@ -223,8 +214,28 @@ func (_ *evaluationHistoryService) ListEvaluationHistory(
 		slices.Reverse(rows)
 	}
 
+	propsvc := ehs.propServiceBuilder(qtx)
+
+	data := make([]OneEvalHistoryAndEntity, 0, len(rows))
+	for _, row := range rows {
+		efp, err := propsvc.EntityForProperties(ctx, row.EntityID, row.ProjectID, ehs.providerManager, qtx)
+		if err != nil {
+			return nil, fmt.Errorf("error fetching entity for properties: %w", err)
+		}
+
+		err = propsvc.RetrieveAllPropertiesForEntity(ctx, efp)
```

Comment:
this is potentially a lot of calls isn't it? Do we actually need up-to-date data here, couldn't we just fetch the entities and properties from cache regardless of whether it's expired or not?

Reply:
We are at a weird point where we don't have everything in cache just yet. The idea is to fetch from cache if it's possible, and persist it if it isn't there. Perhaps I should instead add "options" to this call and rely on the cache even if the data is marked as stale. wdyt?